src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -260,23 +257,7 @@ void Ekf::predictCovariance(const imuSample &imu_delayed)
 
 			for (unsigned index = 0; index < State::mag_B.dof; index++) {
 				unsigned i = State::mag_B.idx + index;
-				nextP(i, i) += mag_B_process_noise;
-			}
-		}
-
-	} else {
-		// keep previous covariance
```

Comment:
How do we handle that now?

Reply:
Isn't it already propagated (upper triangular) in P_new?

https://github.com/PX4/PX4-Autopilot/blob/0eea925087eb86def1a30dba6003e78969fc2ab2/src/modules/ekf2/EKF/python/ekf_derivation/generated/predict_covariance.h#L926-L929

Reply:
You're right. I never thought about it, but only the attitude, vel and pos covariances are affected by the prediction, so this part of the code was useless.

Reply:
Yes, but with that code we're also stopping the propagation of the covariance between the mag and all the other states.